src/grammar.cc

```diff
@@ -655,7 +673,11 @@ void Grammar::put_table_conf(std::ostream &s) {
       hashtable<std::string, Symbol::NT*>::iterator a = tabulated.find(
         *i->second->name);
       if (a == tabulated.end()) {
-        assert(false);
+        // user might not have considered outside NTs to be tabulated in
```

Comment:
does the automated table design consider the NTs of the outside grammar as well when calculating a good design?

Reply:
yes it does :-) Watch out for `outside_multiloop` in the following example. A heuristic table design `-t` would not tabulate this NT, while forcing to tabulate all NTs will of course also create a table for `outside_multiloop`:
```
(cpplint) t490s x86_64 /Daten/Git/jlab/gapc_backward>./gapc testdata/grammar_outside/nodangle.gap --outside_grammar ALL -l 0 -t
Info: 
Grammar has been modified into an outside version.
The following non-terminals will be tabulated (%name indicates linear table, %name% indicates constant tables):
  {dangle, iloop, ml_comps, ml_comps1, outside_dangle, outside_iloop, outside_ml_comps, outside_ml_comps1, outside_strong, outside_struct, outside_weak, strong, struct, weak, } #14 of 25 

resulting in an estimated runtime of:
  22n^3  + 8956n^2  + 2
(cpplint) t490s x86_64 /Daten/Git/jlab/gapc_backward>./gapc testdata/grammar_outside/nodangle.gap --outside_grammar ALL -l 0 --tab-all
Info: 
Grammar has been modified into an outside version.
The following non-terminals will be tabulated (%name indicates linear table, %name% indicates constant tables):
  {dangle, hairpin, iloop, leftB, ml_comps, ml_comps1, multiloop, %outside_axioms%, outside_dangle, outside_hairpin, outside_iloop, outside_leftB, outside_ml_comps, outside_ml_comps1, outside_multiloop, outside_rightB, outside_stack, outside_strong, outside_struct, outside_weak, rightB, stack, strong, struct, weak, } #25 of 25 

resulting in an estimated runtime of:
  22n^3  + 8966n^2  + 4
(cpplint) t490s x86_64 /Daten/Git/jlab/gapc_backward>
```